### PR TITLE
Lodash removal

### PR DESCRIPTION
**Pull request by chazzmoney**
Initial efforts are replacing lodash. This is amateur hour at the moment; all suggestions welcome.

### CODE REVIEW

File: lib/utils/es6_.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+let isArray = function(obj) {
+  return Array.isArray(obj)
+}
+
+module.exports = {
+  each: forEach,
+  forEach: forEach,
+  forIn: forEach,
+  keys: Object.keys,
```
Comment (dbanksdesign) [inline lib/utils/es6_.js:53]: For something like this, could we instead replace instances of `_.keys()` with `Object.keys()`? What do you think?

File: lib/utils/es6_.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      return cpObj;
+    }, {})
+  }
+}
+
+let isString = function(obj) {
+  return typeof obj === 'string' || obj instanceof String
+}
+
+let isArray = function(obj) {
```
Comment (dbanksdesign) [inline lib/utils/es6_.js:45]: Same question as `_.keys()`. What do you think?

File: lib/utils/es6_.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+let forEach = function(obj, f) {
+  Object.keys(obj).forEach((key) => {
+    let value = obj[key]
+    f(value, key)
+  });
+}
+
+let cloneDeep = function(obj) {
```
Comment (dbanksdesign) [inline lib/utils/es6_.js:21]: This method is very clean. ✨

**Review by dbanksdesign — COMMENTED** [review]
❤️ this PR! 2 questions about `_.keys` and `_.isArray`, curious your thoughts

**Comment by dbanksdesign** [comment]
It is fitting that this is PR # 500 😁

File: lib/utils/es6_.js (hunk 1 above)
Comment (didoo) [reply lib/utils/es6_.js:85]: +1

Comment (chazzmoney) [reply lib/utils/es6_.js:246]: Yes, we should do this. My suggestion is to remove _. first, and then switch to native code everywhere else after.

**Comment by chazzmoney** [comment]
@dbanksdesign 

So it is getting close... the few issues left:
1. My replacement for _.pull does not play well with transform/object.js line 86. It causes the tests to hang. Replace '_.pull' with '_es6.pull' to see it in action... Likely I just need to dive deeper on what is being passed in / out
2. Some very hefty work to be done to replace _.template
3. The case-changing stuff is basically copied from lodash and includes its quirks. So.... maybe we want to redo it another way (issues #449 #300 #257)

**Comment by chazzmoney** [comment]
What if we switched from lodash to https://github.com/blakeembrey/change-case ?

**Comment by dbanksdesign** [comment]
I am good swapping one dependency for another is ok, especially if this dependency is more configurable and generally a smaller and more narrow dependency, which this is.

**Comment by chazzmoney** [comment]
moved to lodash-removal-3 #570

**Pull request closed without merging**